Sources/InfomaniakLogin/Networking/InfomaniakNetworkLogin.swift

```diff
@@ -57,13 +57,23 @@ public class InfomaniakNetworkLogin: InfomaniakNetworkLoginable {
     }
 
     public func refreshToken(token: ApiToken, completion: @escaping (ApiToken?, Error?) -> Void) {
+        guard let refreshToken = token.refreshToken else {
+            completion(nil, InfomaniakLoginError.noRefreshToken)
+            return
+        }
+
         var request = URLRequest(url: tokenApiURL)
 
-        let parameterDictionary: [String: Any] = [
+        var parameterDictionary: [String: Any] = [
             "grant_type": "refresh_token",
             "client_id": config.clientId,
-            "refresh_token": token.refreshToken
+            "refresh_token": refreshToken
         ]
+
+        if config.accessType == .none {
+            parameterDictionary["duration"] = "infinite"
```

Comment:
Can we add a Sentry / Matomo to track the migration?

Reply:
As there are no dependencies on the login framework apart from the DI. It would be hard to implement directly on the call. However, we can do it on a per app basis when we call back the app to update the token in the keychain.

Reply:
Ok, perfect, I had the suspicion. Super cool if we can have it in the app.